src/hooks/locationHooks.ts

```diff
@@ -0,0 +1,61 @@
+import { useContext, useEffect, useState } from 'react';
+import * as Location from 'expo-location';
+
+import { SettingsContext } from '../SettingsProvider';
+import { storageHelper } from '../helpers';
+import { LocationSettings } from '../types';
+
+const requestAndFetchPosition = async (
+  setAndSyncLocationSettings: (arg: LocationSettings) => Promise<void>
+) => {
+  const { status } = await Location.requestForegroundPermissionsAsync();
+
+  await setAndSyncLocationSettings({ sortPOIs: status === Location.PermissionStatus.GRANTED });
+
+  if (status === Location.PermissionStatus.GRANTED) {
+    return await Location.getCurrentPositionAsync({});
+  }
+};
+
+export const useLocationSettings = () => {
```

Comment:
got multiple errors/warnings on my android 8.1:

1. [Unhandled promise rejection: Error: Location provider is unavailable. Make sure that location services are enabled.]
2. [Unhandled promise rejection: Error: Encountered an exception while calling native method: Exception occurred while executing exported method setNotificationChannelAsync on module ExpoNotificationChannelManager: Unknown color]

(i do not know if this applies to the code marked, but i wanted to have conversations possible after this post)

Reply:
2. should probably not have anything to do with this. and is more likely to be an issue with push notifications.

1. what is your current version of expo go on your testing device? there seems to have been an issue with google changing stuff, but for me it works fine. new versions of expo go should fix the issue. if that does not fix it, then we need to investigate further.

Reply:
related links:
https://github.com/expo/expo/issues/9377
https://github.com/expo/expo/issues/14248

Reply:
there might be further issues with android: currently the `getCurrentPositionAsync' call does not resolve, on my testing device if i turn off wifi. i am doing further research

Reply:
it might be due to my device/android not properly supporting offline location, as the google maps app will also no longer update the location when wifi is turned off.
as of now, i am unsure if that is depending on the device, or if that is a "feature" of android in general.
do you have any idea?

PS: about the notification thing please look [here](https://github.com/ikuseiGmbH/smart-village-app-app/pull/352#issuecomment-934331027)